test/python/test_ibmqjob.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name,missing-docstring,broad-except
+
+# Copyright 2017 IBM RESEARCH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+"""IBMQJob Test."""
+
+import unittest
+import time
+import numpy
+from scipy.stats import chi2_contingency
+
+from qiskit import (ClassicalRegister, QuantumCircuit, QuantumRegister,
+                    QuantumJob)
+import qiskit._compiler
+from qiskit.backends.ibmq import IBMQProvider
+from .common import requires_qe_access, QiskitTestCase, slow_test
+
+
+class TestIBMQJob(QiskitTestCase):
+    """
+    Test ibmqjob module.
+    """
+
+    @classmethod
+    @requires_qe_access
+    def setUpClass(cls, QE_TOKEN, QE_URL):
+        # pylint: disable=arguments-differ
+        super().setUpClass()
+        # create QuantumCircuit
+        qr = QuantumRegister(2, 'q')
+        cr = ClassicalRegister(2, 'c')
+        qc = QuantumCircuit(qr, cr)
+        qc.h(qr[0])
+        qc.cx(qr[0], qr[1])
+        qc.measure(qr, cr)
+        cls._qc = qc
+        cls._provider = IBMQProvider(QE_TOKEN, QE_URL)
+
+    def test_run(self):
```

Comment:
Can you add another `test_run_device` (or similar name), with `@slow_test`, that runs this same test against a real device? It seems to be failing (tested with `ibmqx4`):

```

self = <test.python.test_ibmqjob.TestIBMQJob testMethod=test_run>

    def test_run(self):
        backend = self._provider.get_backend('ibmqx4')
        qobj = qiskit._compiler.compile(self._qc, backend)
        quantum_job = QuantumJob(qobj, backend, shots=1024, preformatted=True)
        job = backend.run(quantum_job)
>       result = job.result()

test/python/test_ibmqjob.py:59: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
qiskit/backends/ibmq/ibmqjob.py:63: in result
    this_result = self._wait_for_job(timeout=timeout, wait=wait)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <qiskit.backends.ibmq.ibmqjob.IBMQJob object at 0x7f4661bc45f8>, timeout = None, wait = 5

    def _wait_for_job(self, timeout=60, wait=5):
        ....
    
        # Get the results
        api_result = self._api.get_job(job_id)
        job_result_return = []
        for index in range(len(api_result['qasms'])):
>           job_result_return.append({'data': api_result['qasms'][index]['data'],
                                      'status': api_result['qasms'][index]['status']})
E           KeyError: 'data'

qiskit/backends/ibmq/ibmqjob.py:245: KeyError
````